.github/actions/k3s-cluster/action.yaml

```diff
@@ -0,0 +1,74 @@
+# Adjusted from https://github.com/jupyterhub/action-k3s-helm
+---
+name: K3S with Helm
+description: |
+  Install Kubernetes (K3S) and Helm.
+
+inputs:
+  k3s-channel:
+    description: K3S channel (https://update.k3s.io/v1-release/channels)
+    required: false
+    default: ""
+
+outputs:
+  kubeconfig:
+    description: Path to kubeconfig file
+    value: ${{ steps.set-output.outputs.kubeconfig }}
+  k3s-version:
+    description: "Installed k3s version, such as v1.20.0+k3s2"
+    value: "${{ steps.set-output.outputs.k3s-version }}"
+  k8s-version:
+    description: "Installed k8s version, such as v1.20.0"
+    value: "${{ steps.set-output.outputs.k8s-version }}"
+  helm-version:
+    description: "Installed helm version, such as v3.4.2"
+    value: "${{ steps.set-output.outputs.helm-version }}"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup k3s ${{ inputs.k3s-channel }}
+      run: |
+        curl -sfL https://get.k3s.io | INSTALL_K3S_CHANNEL="${{ inputs.k3s-channel }}" sh -s - 
+      shell: bash
+
+    # By providing a kubeconfig owned by the current user with 600 permissions,
+    # kubectl becomes usable without sudo, and helm won't emit warnings about
+    # bloated access to group/world.
+    - name: Prepare a kubeconfig in ~/.kube/config
+      run: |
+        mkdir -p ~/.kube
+        sudo cat /etc/rancher/k3s/k3s.yaml > "$HOME/.kube/config"
+        chmod 600 "$HOME/.kube/config"
+        echo "KUBECONFIG=$HOME/.kube/config" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Setup Helm
+      run: |
+        curl -sf https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
+      shell: bash
+
+    - name: Set version output
+      id: set-output
+      run: |
+        echo "::group::Set version output"
+        echo "::set-output name=kubeconfig::$HOME/.kube/config"
```

Comment:
what is `::set-output` doing? the actual output in the tests are empty

Reply:
it sets the kubeconfig output variable: https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#setting-an-output-parameter